eslint.config.mjs

```diff
@@ -0,0 +1,75 @@
+import tseslint from 'typescript-eslint'
+import globals from 'globals'
+import importPlugin from 'eslint-plugin-import'
+
+/**
+ * @type {import('eslint').Linter.FlatConfig[]}
+ */
+export default [
```

Comment:
I think we can use the `tseslint.config` helper now that things are more stable, and also add a `@ts-check`

Reply:
The issue with adding the `@ts-check` is that some of the plugins don't provide their own declaration files (and neither does ESLint), and I'd rather not add more packages

Reply:
It's a matter of preference, but personally I prefer having a `@ts-check` on by default and a `@ts-expect error incompatible/incorrect type definition` to silence the instances we know works fine, just to prevent accidental mistakes

Reply:
Made the changes as requested